module-system/module-implementations/sov-blob-storage/src/lib.rs

```diff
@@ -0,0 +1,95 @@
+#![deny(missing_docs)]
+
+//! Blob storage module allows to save DA blobs in the state
+
+mod genesis;
+
+use sov_modules_api::{Error, Module, Spec};
+use sov_modules_macros::ModuleInfo;
+use sov_state::{StateValue, WorkingSet};
+
+/// Blob storage contains only address and vector of blobs
+#[derive(ModuleInfo, Clone)]
+pub struct BlobStorage<C: sov_modules_api::Context> {
+    /// The address of blob storage module
+    /// Note: this is address is generated by the module framework and the corresponding private key is unknown.
+    #[address]
+    pub(crate) address: C::Address,
+
+    /// Actual storage of blobs
+    /// stored this way (DA block number, blob)
+    /// TODO: StateVec: https://github.com/Sovereign-Labs/sovereign-sdk/issues/33
+    #[state]
+    pub(crate) blobs: StateValue<Vec<(u64, Vec<u8>)>>,
```

Comment:
Should we limit the size of the `Vec` so it is not unbounded?  This could be done just before insertion to the vec.